src/pycountry/__init__.py

```diff
@@ -6,24 +6,56 @@
 
 import pycountry.db
 
+# importlib.resources is available from Python 3.7
 try:
-    import pkg_resources
+    import importlib.resources
 
-    resource_filename = pkg_resources.resource_filename
-except ImportError:
+    HAS_IMPORTLIB_RESOURCES = True
 
     def resource_filename(package_or_requirement, resource_name):
-        return os.path.join(os.path.dirname(__file__), resource_name)
+        return str(importlib.resources.files(package_or_requirement) / resource_name)
+
+except ImportError:
+    HAS_IMPORTLIB_RESOURCES = False
+
+
+# importlib.metadata is available from Python 3.8
+try:
+    import importlib.metadata
+
+    HAS_IMPORTLIB_METADATA = True
 
-else:
+    def get_version(distribution_name):
+        try:
+            return importlib.metadata.version(distribution_name)
+        except importlib.metadata.PackageNotFoundError:
+            return "n/a"
+except ImportError:
+    HAS_IMPORTLIB_METADATA = False
+
+
+# For python < 3.7, try to use pkg_resources
+if not (HAS_IMPORTLIB_RESOURCES and HAS_IMPORTLIB_METADATA):
     try:
-        __version__ = pkg_resources.get_distribution("pycountry").version
-    except pkg_resources.DistributionNotFound:
-        __version__ = "n/a"
+        import pkg_resources
```

Comment:
Rather than falling back to `pkg_resources` (which is apparently fully deprecated now), it would probably be better to fall back to `importlib_resources` and `importlib_metadata` from PyPI.  They can be added as conditional dependencies for 3.7.

Reply:
It wasn't fully deprecated when I opened the PR. Happy to change it to use the backported packages, but they require Python 3.7+ and `pycountry` still declares support for 3.6. Would we want to keep the `pkg_resources` fallback for 3.6 or wait for #150 to be merged?

Reply:
3.6 is EOL, so I personally wouldn't worry about it :).  I think the local fallbacks should be plenty good enough for 3.6 anyway, and possibly even for 3.7 which would allow having no runtime dependencies.

Reply:
Ugh, annoying. I just noticed `importlib.resources.version` is only available since Python 3.9, so we can't simply rely on `ImportError`s for the whole module. I think t's easier to simply use the pypi packages for all versions. This can be changed once support for 3.9 is dropped.